ED/src/dynamics/rk4_driver.F90

```diff
@@ -760,6 +770,51 @@ subroutine initp2modelp(hdid,initp,csite,ipa,nighttime,wbudget_loss2atm,ebudget_
       ! snow.                                                                              !
       !------------------------------------------------------------------------------------!
       do ico = 1,cpatch%ncohorts
+         !---------------------------------------------------------------------------------!
+         ! First, update variables related with plant hydrodynamics because it can 
+         ! change leaf/wood heat capacity, which will be used later 
+         !---------------------------------------------------------------------------------!
+         if (plant_hydro_scheme > 0) then
+             ! Need to update leaf_water_int and wood_water_int diagnostically
+             ! from changes in heat capacity
+             cpatch%leaf_water_int(ico) = cpatch%leaf_water_int(ico)                    &
```

Comment:
Same as my comment in rk4_derivs.F90.  Mass should be the prognostic variable, not heat capacity.